microcosm_logging/decorators.py

```diff
@@ -15,3 +15,54 @@ class MyClass(object):
     """
     obj.logger = getLogger(obj.__name__)
     return obj
+
+
+class ContextLogger(LoggerAdapter):
+    """
+    Allows for inserting additional context into log records based on the current context.
+    To play nicely with the pythonjsonlogger used for json logs, a ContextLogger will
+    update keys on the existing message object if the message is a dictionary otherwise,
+    it will prepend context information before the log message in brackets.
+
+    """
+    def process(self, msg, kwargs):
+        if isinstance(msg, dict):
+            msg.update(self.extra)
+        else:
+            headers_string = [
```

Comment:
I'm not sure it will work without dynamically changing the associated formatter with each application of the context logger, which seems overly magical.

I think our options are: 
- leave it out of the string logging
- put it in the same way as currently

I think the first 1 makes more sense given we don't want to impose any weird constraints on people's log formatters.